app/presenters/paginator.presenter.js

```diff
@@ -105,26 +105,25 @@ const COMPLEX_END_PAGINATOR = 'end'
  * @param {number} numberOfRecords - the total number of records or results of the thing being paginated
  * @param {number} selectedPageNumber - the page of results selected for viewing
  * @param {string} path - the URL path the paginator should use, for example, `'/system/bill-runs'`
+ * @param {number} currentAmount - The number of items currently being shown (e.g. items on the current page).
+ * @param {string} message - A label appended to the end of the 'Showing x of y' string (e.g. "returns", "results").
  * @param {object} queryArgs - the current query arguments to be added to the pagination links
  *
  * @returns {object} if no pagination is needed just the `numberOfPages` is returned else a `component:` property is
  * also included that can be directly passed to the `govukPagination()` in the view.
  */
-function go(numberOfRecords, selectedPageNumber, path, queryArgs = {}) {
+function go(numberOfRecords, selectedPageNumber, path, currentAmount, message, queryArgs = {}) {
```

Comment:
I would suggest following the pattern of `numberOfPages` and determine `showingMessage` once at the top of the function.

```javascript
  const numberOfPages = Math.ceil(numberOfRecords / DatabaseConfig.defaultPageSize)
  const showingMessage = showingXofY(numberOfRecords, currentAmount, message)

  if (numberOfPages < 2) {
    return { numberOfPages, showingMessage }
  }

  const queryString = _queryString(queryArgs)

  const component = _component(selectedPageNumber, numberOfPages, path, queryString)

  return { component, numberOfPages, showingMessage }
```